javav2/example_code/ecr/src/main/java/com/example/ecr/scenario/ECRActions.java

```diff
@@ -0,0 +1,565 @@
+
+        // Use whenComplete to handle the response or any exceptions.
+        response.whenComplete((resp, ex) -> {
+            if (resp != null) {
+                System.out.println("Repository policy retrieved successfully.");
+            } else {
+                if (ex.getCause() instanceof EcrException) {
+                    EcrException e = (EcrException) ex.getCause();
+                    String errorMessage = "Error getting repository policy for repository: " + repoName + " - " + e.awsErrorDetails().errorMessage();
+                    System.err.println(errorMessage);
+                    e.printStackTrace();
+                } else {
+                    String errorMessage = "Unexpected error occurred: " + ex.getMessage();
+                    System.err.println(errorMessage);
+                    ex.printStackTrace();
+                }
+            }
+        });
+
+        // Wait for the CompletableFuture to complete and return the result.
+        GetRepositoryPolicyResponse result = response.join();
+        return result != null ? result.policyText() : null;
+    }
+    // snippet-end:[ecr.java2.get.repo.policy.main]
+
+    // snippet-start:[ecr.java2.set.repo.policy.main]
+
+    /**
+     * Sets the repository policy for the specified ECR repository.
+     *
+     * @param repoName the name of the ECR repository.
+     * @param iamRole  the IAM role to be granted access to the repository.
+     * @throws InvalidParameterException         if the specified IAM role is invalid.
+     * @throws RepositoryPolicyNotFoundException if the repository policy does not exist.
+     * @throws EcrException                      if there is an unexpected error setting the repository policy.
+     */
+    public void setRepoPolicy(String repoName, String iamRole) {
+        String policyDocumentTemplate = """
+             {
+              "Version" : "2012-10-17",
+              "Statement" : [ {
+                "Sid" : "new statement",
+                "Effect" : "Allow",
+                "Principal" : {
+                  "AWS" : "%s"
+                },
+                "Action" : "ecr:BatchGetImage"
+              } ]
+            }
+             """;
+
+        String policyDocument = String.format(policyDocumentTemplate, iamRole);
+        SetRepositoryPolicyRequest setRepositoryPolicyRequest = SetRepositoryPolicyRequest.builder()
+            .repositoryName(repoName)
+            .policyText(policyDocument)
+            .build();
+
+        CompletableFuture<SetRepositoryPolicyResponse> response = getAsyncClient().setRepositoryPolicy(setRepositoryPolicyRequest);
+        response.whenComplete((resp, ex) -> {
+            if (resp != null) {
+                System.out.println("Repository policy set successfully.");
+            } else {
+                if (ex.getCause() instanceof InvalidParameterException) {
```

Comment:
Would it be better to set `ex.getCause()` to a variable and reuse it?

Reply:
I made the modification to use a variable.